src/main/java/com/google/sps/app/servlets/SignUp.java

```diff
@@ -13,15 +12,20 @@
 import com.google.cloud.datastore.Entity;
 import com.google.cloud.datastore.FullEntity;
 import com.google.cloud.datastore.KeyFactory;
-import com.google.gson.Gson;
 import com.google.sps.util.UUIDs;
 import com.google.sps.util.Validator;
 
 @WebServlet("/SignUp")
 public class SignUp extends HttpServlet{
 
-    private static final String USER_NAME_PARAM =
-        "text-input-user-name";
+    private static final String USER_NAME_PARAM = "text-input-user-name";
+    // Formatted string for validation error messages
+    private static final String VALIDATOR_ERROR_MESSAGE = "<p class=\"error\">Invalid Username: %s<p>";
+    // Validation error message discriptions
+    private static final String IMPROPER_CHARACTERS = VALIDATOR_ERROR_MESSAGE + "Improper characters. Please use letters and numbers only.";
+    private static final String USERNAME_LENGTH = VALIDATOR_ERROR_MESSAGE + "Usernames must be between 1 and 64 characters";
+    // Taken user name error message
+    private static final String USERNAME_TAKEN = "<p class=\"error\">Username Taken: <a href=\"/LogIn\">Login</a></p>";
     
```

Comment:
If these are similar to the signup ones,move it to a errors.java class. No need to do it in this PR.

Reply:
Good Idea